e2e/tests/008.sh

```diff
@@ -28,8 +29,47 @@ source "${LIBDIR}/k8s.sh"
 
 kubeconfig="$HOME/.kube/config"
 
+function k8s_upf_scale_test {
+    local current_cpu=$1
+    local current_memory=$2
+    local after_scale_cpu=$3
+    local after_scale_memory=$4
+    
+    echo "Comparing the new CPU/Memory before and after scaling"
+
+    if [ "$after_scaling_cpu" -le  "$current_cpu" ] && [ "$after_scale_memory" -le  "$current_memory" ]; then
+        echo "UPF POD Scaling Failed"
+        exit 1
+    else
+        echo "UPF Pod Scaling Successful"
+    fi
+    
+    exit 0
+}
```

Comment:
This function can be implemented in a more generic way:
```suggestion
function _check_scale {
    local metric=$1
    local previous=$2
    local current=$3
    
    echo "Comparing the new $metric after scaling"

    if [ "$previous" -le  "$current" ]; then
        echo "$metric scaling Failed"
        exit 1
    fi
    echo "$metric Pod Scaling Successful"
}
```

So the calling function will be:

```bash
_check_scale "CPU" $current_cpu $after_scaling_cpu
_check_scale "Memory" $current_memory $after_scaling_memory
```

Reply:
Ok sure, will change it to this logic.